File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if i == "(":
            stack.append(i)
        elif i == ")" and len(stack) == 0:
            return False
        elif i == ")":
            stack.pop()
    return True if len(stack) == 0 else False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
import java.io.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '(')
                stack.push('(');
            else if((!stack.empty())&&(s.charAt(i) == ')')&& (stack.peek() == '('))
                stack.pop();
            else
                return false;
        }
        if (stack.empty())
            return true;
        else
            return false;
        
    }
}